### PR TITLE
Update `spice_cloud` - connect to cloud api

### PR DESCRIPTION
**Pull request by ewgenius**
Refactored `spice_cloud` extension:
- calls spice cloud api, to get current app and metrics dataset
- updated spice_cloud config: optional `api_url` (for testing), `control_plane_enabled` to enable control plane runtime functionality

Examples:

- if `spice_cloud` extension not configured in spicepod - runtime will use just local metrics table

- default extension config:

```yaml
extensions:
  spice_cloud:
    enabled: true
```

or 

```yaml
extensions:
  spice_cloud: {}
```

- custom spice api url:

```yaml
extensions:
  spice_cloud:
    params:
      api_url: https://dev-data.spiceai.io
```

CleanShot 2024-05-28 at 11 16 30@2x

### CODE REVIEW

File: crates/spice_cloud/src/lib.rs [hunk 1]
```diff
@@ -34,6 +36,20 @@ pub enum Error {
...
+        source: Box<dyn std::error::Error + Sync + Send>,
+    },
+
+    #[snafu(display("Spice Cloud secret not found"))]
+    SpiceSecretNotFound {},
+
+    #[snafu(display("Spice Cloud api_key not provided"))]
+    SpiceApiKeyNotFound {},
+
+    #[snafu(display("Unable to connect to Spiec Cloud: {source}"))]
```
Comment (phillipleblanc) [inline crates/spice_cloud/src/lib.rs]:
```suggestion
    #[snafu(display("Unable to connect to Spice Cloud: {source}"))]
```

File: crates/spice_cloud/src/lib.rs [hunk 2]
```diff
@@ -45,6 +61,110 @@ impl SpiceExtension {
     pub fn new(manifest: ExtensionManifest) -> Self {
         SpiceExtension { manifest }
     }
+
+    fn control_plane_enabled(&self) -> bool {
+        self.manifest
+            .params
+            .get("control_plane_enabled")
+            .unwrap_or(&"false".to_string())
+            == "true"
```
Comment (phillipleblanc) [inline crates/spice_cloud/src/lib.rs]: This should probably be a case-insensitive compare

**Review by phillipleblanc — DISMISSED** [review]

File: crates/spice_cloud/src/lib.rs (hunk 2 above)
Comment (ewgenius) [reply crates/spice_cloud/src/lib.rs]: fixed, thanks

**Review by phillipleblanc — DISMISSED** [review]

File: crates/runtime/src/lib.rs [hunk 3]
```diff
@@ -231,6 +231,7 @@ impl Runtime {
                 tracing::warn!("Failed to initialize extension {extension_name}: {err}");
             } else {
                 extensions.push(extension);
+                tracing::info!("Loaded extension {extension_name}");
```
Comment (y-f-u) [inline crates/runtime/src/lib.rs:235]: HOLD THE MERGE FOR NOW

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**